### ISSUE TITLE
Script do not work under channel Videos section (/user/foo/videos)

### ISSUE THREAD

**Issue by gkrystev** [EvHaus/youtube-hide-watched#13]
Ok, I did a little debugging and manage to figure out something.

So, the script does not work under channel Videos section (location ending with /videos).

In $(findWatchedElements()).each loop, because the location is not '/feed/subscriptions' it searches for class '.ytd-video-renderer', but in this page there are not such elements. Instead, it needs to search for class '.ytd-grid-renderer' as in '/feed/subscriptions'.
Not sure do I explain it right, as my JavaScript knowledge is very limited.

So, replacing this:
```
 if (window.location.href.indexOf('/feed/subscriptions') > 0) {
                row = item.closest('#grid-container');
                gridItem = item.closest('.ytd-grid-renderer');
            } else {
                row = item.closest('.ytd-video-renderer');
            }
```
with this:
```
 if (window.location.href.indexOf('/feed/subscriptions') > 0) {
                row = item.closest('#grid-container');
                gridItem = item.closest('.ytd-grid-renderer');
            } else if (window.location.href.match(/.*\/videos$/)) {
                row = item.closest('.ytd-grid-renderer');
            } else {
                row = item.closest('.ytd-video-renderer');
            }
```
make it works again.

Maybe it can be done better, but as I said, not my language.

**Comment by EvHaus** [comment]
Nice catch. Your code looks good. Fixed in v2.4

**Comment by pejotigrek** [comment]
I don't want to double anything, so forgive me for digging up this old issue - script works fine with "clean" url of the videos section - but if only you change some settings [like: sorting etc.] and arguments goes to url, it doesn't work anymore.

example url [I'm catching up old Casey's vlogs ;)]:
works with: https://www.youtube.com/user/caseyneistat/videos
doesn't work: https://www.youtube.com/user/caseyneistat/videos?sort=da&view=0&flow=grid

**Comment by EvHaus** [comment]
Good catch @pejotigrek! Fixed in v2.7

**Comment by pejotigrek** [comment]
wow, @EvHaus thanks for a quick update! but now the button to turn on/off disapeared.. I didn't changed anything in my Chrome's config or scripts, do you think it's just a coincidence?
